receiver.py

```diff
@@ -22,17 +23,23 @@ def __init__(self, gen, filename, header):
 
     def data_lines(self):
         # write header line out to file
-        print('writing headers')
-        with open(self.filename, 'w') as outfile:
-            outfile.write("{}\r\n".format(self.header))
 
-        for line in self.gen.data_lines():
-            # write element to file
-            with open(self.filename, 'a') as outfile:
-                outfile.write(str(line).replace('(','').replace(')','').replace('None','') + '\n')
-            # re-yield element
-            yield line
 
+        try:
+            print('writing headers')
+            with open(self.filename, 'w') as outfile:
+                outfile.write("{}\r\n".format(self.header))
+
+            for line in self.gen.data_lines():
+                # write element to file
+                with open(self.filename, 'a') as outfile:
+                    outfile.write(str(line).replace('(','').replace(')','').replace('None','') + '\n')
+                # re-yield element
+                yield line
+        except (OSError, serial.SerialException):
+            #catch exception if xbee is unplugged, and try to reconnect
+            print "Xbee disconnected!"
```

Comment:
We are aiming for python 3 compatibility, so prints should use python 3 syntax. (ie. you need brackets)